docs/testing/webdriverio/02-unit-testing.md

```diff
@@ -0,0 +1,69 @@
+
+### Test Setup
+
+To resemble how your component is being used as close as possible to reality we need to render it into an actual DOM tree. WebdriverIO provides a helper package for this that you can use called `@wdio/browser-runner/stencil`. It exports a `render` method that allows us to mount our component to the DOM.
+
+For example, given the following component:
+
+```ts reference title="/src/components/my-component/my-component.tsx"
+https://github.com/webdriverio/component-testing-examples/blob/main/stencil-component-starter/src/components/my-component/my-component.tsx
+```
+
+We import the component into our test to render it in the browser:
+
+```ts reference title="/src/components/my-component/my-component.test.tsx"
+https://github.com/webdriverio/component-testing-examples/blob/main/stencil-component-starter/src/components/my-component/my-component.test.tsx#L2-L18
+```
+
+If your component under test uses other Stencil components make sure you add these to the `components` list to. For example, let's say `ComponentA` uses `ComponentB` which also imports `ComponentC` and `ComponentD`. In this case you will have to import and pass in all components you like to have rendered, e.g.:
+
+```ts
+render({
+    components: [ComponentA, ComponentB, ComponentC, ComponentD],
+    template: () => <component-a first="Stencil" last="'Don't call me a framework' JS" />
+});
+```
+
+While this seems tedious at first, it gives you the flexibility to leave out components that are not relevant for your test, or have side effects that can cause flakiness.
+
+Find more information about the `render` method option and its return object in the WebdriverIO [documentation](https://webdriver.io/docs/component-testing/stencil#render-options).
+
+### Handle Asynchronicity
+
+Instead of directly working on DOM objects, with WebdriverIO you are interacting with references of DOM nodes and interact through WebDriver commands that are async. Make sure you always use an `await` to ensure that all commands and assertion are executed sequentially.
+
+:::info
+
+Missing an `await` can be a simple oversight and can cause us long hours of debugging. To avoid this and ensure promises are handled properly, it is recommended to use an ESLint rule called [`require-await`](https://eslint.org/docs/latest/rules/require-await).
+
+:::
+
+### Matchers
+
+WebdriverIO provides their own matchers to assert an element in various ways. We recommend to use them over synchronous matchers like `toBe` or `toEqual` as they allow for retries and make your tests more resilient against flakiness.
+
+For example, instead of asserting the content of a component like this:
+
+```ts
+expect(await $('my-component').getText())
+    .toBe(`Hello, World! I'm Stencil 'Don't call me a framework' JS`)
+```
+
+It is better to use WebdriverIOs matchers for asserting text:
+
+```ts
+await expect($('my-component'))
+    .toHaveText(`Hello, World! I'm Stencil 'Don't call me a framework' JS`)
+```
```

Comment:
is this preferable just because it will retry if there's no textContent yet, or something along those lines?